src/main/java/domain/board/BoardGenerator.java

```diff
@@ -0,0 +1,156 @@
+package domain.board;
+
+import static domain.board.Board.END_COLUMN_INDEX;
+import static domain.board.Board.END_ROW_INDEX;
+import static domain.board.Board.START_COLUMN_INDEX;
+import static domain.board.Board.START_ROW_INDEX;
+
+import domain.piece.PieceType;
+import domain.piece.Byeong;
+import domain.piece.Cha;
+import domain.piece.Ma;
+import domain.piece.Piece;
+import domain.piece.Po;
+import domain.piece.Sa;
+import domain.piece.Sang;
+import domain.piece.Team;
+import domain.piece.Wang;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import view.SangMaOrderCommand;
+
+public class BoardGenerator {
+
+    public Board generateBoard(SangMaOrderCommand hanSangMaOrderCommand, SangMaOrderCommand choSangMaOrderCommand) {
+        Map<Point, Node> nodeByPoint = initializeNodesAndEdges();
+        Map<Node, Piece> pieceByNode = initializePiecePosition(nodeByPoint,
+                hanSangMaOrderCommand,
+                choSangMaOrderCommand);
+        return new Board(pieceByNode, nodeByPoint);
+    }
+
+    public Map<Point, Node> initializeNodesAndEdges() {
+        Map<Point, Node> nodeByPoint = new HashMap<>();
+
+        for (int row = START_ROW_INDEX; row <= END_ROW_INDEX; row++) {
+            for (int column = START_COLUMN_INDEX; column <= END_COLUMN_INDEX; column++) {
+                Point point = Point.of(row, column);
+                Node currentNode = new Node(point);
+                nodeByPoint.put(point, currentNode);
+            }
+        }
+
+        for (int row = START_ROW_INDEX; row <= END_ROW_INDEX; row++) {
+            for (int column = START_COLUMN_INDEX; column <= END_COLUMN_INDEX; column++) {
+                Point point = Point.of(row, column);
+                Node currentNode = nodeByPoint.get(point);
+                currentNode.addAllEdges(createEdges(row, column, nodeByPoint));
+            }
+        }
+
+        return nodeByPoint;
+    }
+
+    private List<Edge> createEdges(int row, int column, Map<Point, Node> nodeByPoint) {
+        List<Edge> edges = new ArrayList<>();
+        for (Direction direction : List.of(Direction.UP, Direction.DOWN, Direction.LEFT, Direction.RIGHT)) {
```

Comment:
`List.of(Direction.UP, Direction.DOWN, Direction.LEFT, Direction.RIGHT)` 와 같은 경우는 자주 사용될 수도 있을 것 같은 List일 것 같은데요. 

이런 경우에는 Direction에 상수로 정의해두고 사용해볼 수도 있을 것 같아요. 

```suggestion
        for (Direction direction : Direction.VERTICALS) {
```

Reply:
그러네요! Cha와 Po에서도 사용되어서 Direction에 상수로 정의해두면 재사용하기 좋을 것 같아요. 수정했습니다 😄 
```java
public enum Direction {

    UP(-1, 0),
    DOWN(1, 0),
    LEFT(0, -1),
    RIGHT(0, 1),
    UP_LEFT(-1, -1),
    UP_RIGHT(-1, 1),
    DOWN_LEFT(1, -1),
    DOWN_RIGHT(1, 1),
    ;

    public static final List<Direction> VERTICALS = List.of(UP, DOWN, LEFT, RIGHT);
    ...
```